### ISSUE TITLE
Unable to build a project with amplify 10.3.0 

### ISSUE THREAD

**Issue by kouz75** [aws-amplify/amplify-cli#11211]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

14.x

### Amplify CLI Version

10.3.0

### What operating system are you using?

macos

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No manual changes made

### Amplify Categories

Not applicable

### Amplify Commands

Not applicable

### Describe the bug

deploy an old project  with amplify console and amplify cli 10.3.0 will failed  
more detail in codebuild log

deploy with version 10.2.x works fine.

run localy, "amplify push" with version 10.3.0  will work.

### Reproduction steps

every thing works fine with a new project ...

### GraphQL schema(s)

-

### Project Identifier

ed57bb8a46476511162afa8497d78147

### Log output

<details>

```

2022-10-18T21:35:58.642Z [INFO]: [0mBackend environment dev found in Amplify Console app: myproject[0m
2022-10-18T21:36:00.727Z [WARNING]: - Fetching updates to backend environment: dev from the cloud.
2022-10-18T21:36:02.485Z [WARNING]: - Building resource api/myprojectcontactapi549
2022-10-18T21:36:02.502Z [WARNING]: - Building resource api/AdminQueries
2022-10-18T21:36:02.516Z [WARNING]: - Building resource api/myprojectgraphql
2022-10-18T21:36:05.319Z [INFO]: ✅ GraphQL schema compiled successfully.
                                 Edit your schema at /codebuild/output/src490452204/src/myproject/amplify/backend/api/myprojectgraphql/schema.graphql or place .graphql files in a directory at /codebuild/output/src490452204/src/myproject/amplify/backend/api/myprojectgraphql/schema
2022-10-18T21:36:05.322Z [WARNING]: - Building resource api/myapi
....
2022-10-18T21:36:05.376Z [WARNING]: - Building resource auth/authressource
2022-10-18T21:36:05.391Z [WARNING]: - Building resource auth/userPoolGroups
2022-10-18T21:36:05.453Z [WARNING]: - Building resource storage/mystorage
2022-10-18T21:36:18.220Z [WARNING]: ✔ Successfully pulled backend environment dev from the cloud.
2022-10-18T21:36:18.433Z [WARNING]: ✖ There was an error initializing your environment.
2022-10-18T21:36:18.914Z [INFO]: 🛑 Failed to pull the backend.
2022-10-18T21:36:18.915Z [INFO]: File at path: '/codebuild/output/src490452204/src/myproject/amplify/backend/auth/myproject0e23e38c0e23e38c/build/parameters.json' does not exist
                                 Learn more at: https://docs.amplify.aws/cli/project/troubleshooting/
2022-10-18T21:36:18.915Z [WARNING]: - Creating Zip
2022-10-18T21:36:19.008Z [INFO]: 
2022-10-18T21:36:19.008Z [INFO]: ✅ Report saved: /tmp/myproject/report-1666128978926.zip
2022-10-18T21:36:19.008Z [WARNING]: - Sending zip
2022-10-18T21:36:19.008Z [INFO]: File at path: '/codebuild/output/src490452204/src/myproject/amplify/backend/amplify-meta.json' does not exist
2022-10-18T21:36:19.010Z [WARNING]: ✖ Sending zip
2022-10-18T21:36:19.152Z [ERROR]: !!! Build failed
2022-10-18T21:36:19.152Z [ERROR]: !!! Non-Zero Exit Code detected

```

</details>

**Comment by liamstar97** [comment]
I'm having this issue as well with amplify version 10.3.0 when I clone my project onto my local machine and try to initialize the backend environment using `amplify init`. This also occurs when my project tries to build in the cloud whenever CD is triggered on commits to git. When I downgrade to version 10.2.0 it works on my local machine. I'm currently trying to find a way to downgrade the version of amplify the console is using online.

**Comment by kolodi** [comment]
I confirm the same issue when building from GIT in CI. Downgrading to previously used 10.0.0 solved the issue.

**Comment by bergmorten** [comment]
I've the same error

**Comment by zackspear** [comment]
`amplify push` worked just fine for me but when pushing to my Github repo it would trigger the amplify hosting to rebuild. 

But it would fail with
```
File at path: '/codebuild/output/src031203168/src/auth/amplify/backend/auth/<resourceName1>/build/parameters.json' does not exist
```
I landed here and the aforementioned workaround worked for me on my `main` branch front-end build but only on the first try after adding the workaround to the building settings `amplify.yml`.

A subsequent deployment attempt resulted in a different error of
```
ENOENT: no such file or directory, lstat '/codebuild/output/src065924861/src/auth/amplify/backend/function/<resourceName2>/src'
```

Sure I could create another similar workaround here but it feels like a whack-a-mole solution for a backend with a lot of resources.

Project Identifier: `df3110bc24ac6a0ad4ef8fc33826839f`

Hoping for a solve soon. Happy to provide any additional information.

**Comment by sammartinez** [comment]
Just as a callout, this affected a project I am working on, is there anyway to avoid this in the future? @renebrandel

**Comment by bergmorten** [comment]
Do you have an ETA when a new amplify-cli will be released or git build is starting to work again?

**Comment by zackspear** [comment]
10.3.1 was released earlier today. It does not solve the front-end build for our app. 

Since this issue started it broke our `amplify custom` config we had to write to link the Cognito user migration to a lambda function – because for some reason that's not supported out of the box with `amplify-cli`. We then realized our `amplify custom` overwrote the default `amplify auth` triggered functions like PreSignUp and PostConfirmation that we had setup. So our `amplify custom` grew to fill in more blanks for lack of support in `amplify-cli`.

We struggled to figure that out and this bug screwed it all up. And finding more and more settings in Cognito that need tweaking but aren't available in `amplify-cli`. So having to write more and more "hacky" code to fill holes this tool has – whether through `amplify custom` or `overrides.ts`.

I'm ranting into the void because it's been so frustrating using Amplify. Nothing but a headache and a battle from the start. Dropping our use of `amplify-cli` in favor of using Terraform - which I wish we had done from the start.

Good luck to any one that uses this in a major production app with paying customers.
